src/components/ServiceRequestForm.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import Categories from "./services/categories.json";
+import * as Yup from "yup";
+import ErrorMsg from "./errormsg";
+import FormContainer from './FormContainer';
+
+const getSubCategories = (categories, categoryId) => {
+	var category = categories.find(category => category.id === categoryId);
+	return category ? category.types : [];
+};
+
+const ServiceRequestForm = props => {
+
+	const [categories] = useState(Categories);
+	const [subCategories, setSubCategories] = useState([]);
+
+	const handleServiceRequestChange = (changeEvent, setFieldValue) => {
+		const { value } = changeEvent.currentTarget;
+		const subCategories = getSubCategories(categories, parseInt(value));
+		setFieldValue(changeEvent.currentTarget.name, changeEvent.currentTarget.value)
```

Comment:
not sure why you are doing this?

Reply:
when user selects the item from the select box on onchange event that selected item does not stay selected unless I do setFieldValue inside handleServiceRequestChange(). I could not find a better way to do it . Please let me know how can i handle this logic. That is why I was passing props.setFieldValue in onChange event .

Reply:
used martys solutions for this issues

Reply:
switched to component RequestTypeField to handle this.